test/verifiedTests/base/testSolvers/testVerifyCobraProblem.m

```diff
@@ -0,0 +1,242 @@
+assert(res == 1);
+
+% remove csense:
+problem = rmfield(problem,'csense');
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid csense
+problem.csense = [1;1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid csense
+problem.csense = [1,1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid csense
+problem.csense = ['E';'E'];
+res = verifyCobraProblem(problem);
+assert(res == 1);
+
+% remove lb:
+problem = rmfield(problem,'lb');
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid lb
+problem.lb = ['E';'E'];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid lb
+problem.lb = [1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid lb (NaN)
+problem.lb = [-10;NaN];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid lb (wrong size)
+problem.lb = [-10,-10];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add valid lb 
+problem.lb = [-10;-10];
+res = verifyCobraProblem(problem);
+assert(res == 1);
+
+
+% remove ub:
+problem = rmfield(problem,'ub');
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid ub
+problem.ub = ['E';'E'];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid ub
+problem.ub = [1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid ub (NaN)
+problem.ub = [-10;NaN];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid ub (wrong size)
+problem.ub = [-10,-10];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid ub (ub < lb)
+problem.ub = [-20;-20];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add valid ub 
+problem.ub = [10;10];
+res = verifyCobraProblem(problem);
+assert(res == 1);
+
+% remove c:
+problem = rmfield(problem,'c');
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid c
+problem.c = ['E';'E'];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid c
+problem.c = [1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid c (wrong size)
+problem.c = [-10,-10];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid c (wrong size)
+problem.c = [-10;NaN];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add valid c - this now is finally a valid Problem.
+problem.c = [1;1];
+res = verifyCobraProblem(problem);
+assert(res == 1);
+
+%Add invalid F
+problem.F = ['EE';'EE'];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid F
+problem.F = [1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid F (wrong size)
+problem.F = [-10,-10];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+%Add invalid F - other dimension
+problem.F = [1;1];
+res = verifyCobraProblem(problem);
+assert(res == -1);
+
+%Add valid F - this now is finally a valid Problem.
+problem.F = [1,0;0,1];
+res = verifyCobraProblem(problem);
+assert(res == 1);
+
+%Add invalid vartype
```

Comment:
Would it be possible to eventually write this test differently, i.e. with a loop?

Reply:
not really, as all the different settings are not necessarily the same for the individual fields.
Some are matricies, some are vectors, some need chars some need doubles etc pp...

Reply:
ok ... I just thought as the tests look quite repetitive.